crates/dt-workspace/src/lib.rs

```diff
@@ -0,0 +1,52 @@
+use std::path::PathBuf;
+
+use config::{Config, ConfigError};
+use thiserror::Error;
+
+pub mod config;
+mod search;
+
+#[derive(Debug)]
+pub struct Workspace {
+    path: PathBuf,
+
+    config: Config,
```

Comment:
```suggestion
    pub path: PathBuf,
    pub config: Config,
```

These should be public for API consumers to use

Reply:
FYI, `mod tests` makes a submodule of this so it can access private items from its parents. No need to add an integration test for this though.